synapse/storage/databases/main/state.py

```diff
@@ -133,6 +146,50 @@ def get_room_version_id_txn(self, txn: LoggingTransaction, room_id: str) -> str:
 
         return room_version
 
+    async def get_metadata_for_events(
+        self, event_ids: Collection[str]
+    ) -> Dict[str, EventMetadata]:
+        """Get some metadata (room_id, type, state_key) for the given events.
```

Comment:
I agree this isn't a great name. What I really want to call it is "pull out basic info about the event without having to pull out and parse everything"